sgl-kernel/src/sgl-kernel/csrc/trt_reduce_internal.cuh

```diff
@@ -0,0 +1,111 @@
+// reference: https://github.com/NVIDIA/TensorRT-LLM/blob/release/0.14/cpp/tensorrt_llm/plugins/ncclPlugin/allreducePlugin.cpp
+/*
+ * Copyright (c) 2022-2024, NVIDIA CORPORATION.  All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+#include <cuda_fp16.h>
+#include <stdint.h>
+#include <torch/all.h>
+
+#define FatalError(s)                                                          \
```

Comment:
Can `FatalError` and `CHECKCUDA` be implemented in something like `error.h`? Other kernels will also use them, and there's some duplicate code with the initial warp reduce.
ref https://github.com/NVIDIA/cuCollections/blob/dev/include/cuco/utility/error.hpp